### PR TITLE
AArch64: Use CASAL for Unsafe#compareAndSwap

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
Use CASAL instruction for Unsafe#compareAndSwap if FEAT_LSE is available.

Signed-off-by: Akira Saitoh <saiaki@jp.ibm.com>

### CODE REVIEW

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -3360,59 +3359,84 @@ genCAS(TR::Node *node, TR::CodeGenerator *cg, TR_ARM64ScratchRegisterManager *sr
...
-   generateTrg1MemSrc1Instruction(cg, op, node, resultReg, new (cg->trHeapMemory()) TR::MemoryReference(addrReg, (int32_t)0, cg), newVReg);
-   generateCompareBranchInstruction(cg, TR::InstOpCode::cbnzx, node, resultReg, loopLabel);
+      TR::LabelSymbol *loopLabel = generateLabelSymbol(cg);
+      generateLabelInstruction(cg, TR::InstOpCode::label, node, loopLabel);
+      if (createDoneLabel)
+         {
+         doneLabel = generateLabelSymbol(cg);
+         }
+      /*
+      * Generating the same instruction sequence as __sync_bool_compare_and_swap
```
Comment (knn-k) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:3392]: Please align the position of `*` in this comment.

Comment (Akira1Saitoh) [reply runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: Fixed the indentation of this comment.

**Comment by knn-k** [comment]
jenkins test sanity alinux64 jdk11

**Comment by knn-k** [comment]
jenkins test extended.functional alinux64 jdk11

**Review by knn-k — APPROVED** [review]
LGTM.

**Pull request merged**